### PR TITLE
Switch to object type (version 1.0.0)

### LINKED ISSUE #64: BadStatusLine from requests made to the Wink API
I occasionally see the below error during startup of Home-Assistant. I have included two different runs each got the same error, but at a different point during startup. The error happens when the Wink API is queried. Right after these two attempts the third was successful. 

Not sure if this is fixable here, if we are doing what we should already and blowing up? Is this an issue that requests should handle better? Are there any other libraries that could be used besides requests that may work better/not receive this error?

I probably see this once in maybe 20 reboots? Although in this case it happened three times in a row.

### This is during startup of Home-Assistant (nothing from wink was added)
```bash
17-01-02 17:17:37 homeassistant.bootstrap: Error during setup of component wink
Traceback (most recent call last):
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 594, in urlopen
    chunked=chunked)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 391, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 387, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.4/http/client.py", line 1227, in getresponse
    response.begin()
  File "/usr/lib/python3.4/http/client.py", line 386, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.4/http/client.py", line 356, in _read_status
    raise BadStatusLine(line)
http.client.BadStatusLine: ''

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/adapters.py", line 423, in send
    timeout=timeout
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 643, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/util/retry.py", line 334, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/packages/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 594, in urlopen
    chunked=chunked)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 391, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 387, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.4/http/client.py", line 1227, in getresponse
    response.begin()
  File "/usr/lib/python3.4/http/client.py", line 386, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.4/http/client.py", line 356, in _read_status
    raise BadStatusLine(line)
requests.packages.urllib3.exceptions.ProtocolError: ('Connection aborted.', BadStatusLine("''",))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/homeassistant-0.36.0.dev0-py3.4.egg/homeassistant/bootstrap.py", line 151, in _async_setup_component
    None, component.setup, hass, config)
  File "/usr/lib/python3.4/asyncio/futures.py", line 358, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.4/asyncio/tasks.py", line 297, in _wakeup
    future.result()
  File "/usr/lib/python3.4/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.4/concurrent/futures/thread.py", line 54, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/homeassistant-0.36.0.dev0-py3.4.egg/homeassistant/components/wink.py", line 82, in setup
    pywink.get_subscription_key(),
  File "/home/w1ll1am/.virtualenvs/hass_dev/config/deps/pywink/api.py", line 190, in get_subscription_key
    response_dict = wink_api_fetch()
  File "/home/w1ll1am/.virtualenvs/hass_dev/config/deps/pywink/api.py", line 204, in wink_api_fetch
    response = requests.get(arequest_url, headers=API_HEADERS)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/api.py", line 70, in get
    return request('get', url, params=params, **kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/api.py", line 56, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/sessions.py", line 488, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/sessions.py", line 609, in send
    r = adapter.send(request, **kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/adapters.py", line 473, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', BadStatusLine("''",))
```

### This is from a different startup (everything but switches and locks were added)
```bash
17-01-02 17:26:13 homeassistant.components.lock: Error while setting up platform wink
Traceback (most recent call last):
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 594, in urlopen
    chunked=chunked)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 391, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 387, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.4/http/client.py", line 1227, in getresponse
    response.begin()
  File "/usr/lib/python3.4/http/client.py", line 386, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.4/http/client.py", line 356, in _read_status
    raise BadStatusLine(line)
http.client.BadStatusLine: ''

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/adapters.py", line 423, in send
    timeout=timeout
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 643, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/util/retry.py", line 334, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/packages/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 594, in urlopen
    chunked=chunked)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 391, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 387, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.4/http/client.py", line 1227, in getresponse
    response.begin()
  File "/usr/lib/python3.4/http/client.py", line 386, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.4/http/client.py", line 356, in _read_status
    raise BadStatusLine(line)
requests.packages.urllib3.exceptions.ProtocolError: ('Connection aborted.', BadStatusLine("''",))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/homeassistant-0.36.0.dev0-py3.4.egg/homeassistant/helpers/entity_component.py", line 150, in _async_setup_platform
    entity_platform.add_entities, discovery_info
  File "/usr/lib/python3.4/asyncio/futures.py", line 358, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.4/asyncio/tasks.py", line 297, in _wakeup
    future.result()
  File "/usr/lib/python3.4/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.4/concurrent/futures/thread.py", line 54, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/homeassistant-0.36.0.dev0-py3.4.egg/homeassistant/components/lock/wink.py", line 18, in setup_platform
    add_devices(WinkLockDevice(lock, hass) for lock in pywink.get_locks())
  File "/home/w1ll1am/.virtualenvs/hass_dev/config/deps/pywink/api.py", line 146, in get_locks
    return get_devices(device_types.LOCK)
  File "/home/w1ll1am/.virtualenvs/hass_dev/config/deps/pywink/api.py", line 215, in get_devices
    response_dict = wink_api_fetch()
  File "/home/w1ll1am/.virtualenvs/hass_dev/config/deps/pywink/api.py", line 204, in wink_api_fetch
    response = requests.get(arequest_url, headers=API_HEADERS)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/api.py", line 70, in get
    return request('get', url, params=params, **kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/api.py", line 56, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/sessions.py", line 488, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/sessions.py", line 609, in send
    r = adapter.send(request, **kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/adapters.py", line 473, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', BadStatusLine("''",))
17-01-02 17:26:14 homeassistant.components.switch: Error while setting up platform wink
Traceback (most recent call last):
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 594, in urlopen
    chunked=chunked)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 391, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 387, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.4/http/client.py", line 1227, in getresponse
    response.begin()
  File "/usr/lib/python3.4/http/client.py", line 386, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.4/http/client.py", line 356, in _read_status
    raise BadStatusLine(line)
http.client.BadStatusLine: ''

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/adapters.py", line 423, in send
    timeout=timeout
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 643, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/util/retry.py", line 334, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/packages/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 594, in urlopen
    chunked=chunked)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 391, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 387, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.4/http/client.py", line 1227, in getresponse
    response.begin()
  File "/usr/lib/python3.4/http/client.py", line 386, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.4/http/client.py", line 356, in _read_status
    raise BadStatusLine(line)
requests.packages.urllib3.exceptions.ProtocolError: ('Connection aborted.', BadStatusLine("''",))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/homeassistant-0.36.0.dev0-py3.4.egg/homeassistant/helpers/entity_component.py", line 150, in _async_setup_platform
    entity_platform.add_entities, discovery_info
  File "/usr/lib/python3.4/asyncio/futures.py", line 358, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.4/asyncio/tasks.py", line 297, in _wakeup
    future.result()
  File "/usr/lib/python3.4/asyncio/futures.py", line 274, in result

    raise self._exception
  File "/usr/lib/python3.4/concurrent/futures/thread.py", line 54, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/homeassistant-0.36.0.dev0-py3.4.egg/homeassistant/components/switch/wink.py", line 20, in setup_platform
    for switch in pywink.get_powerstrip_outlets():
  File "/home/w1ll1am/.virtualenvs/hass_dev/config/deps/pywink/api.py", line 162, in get_powerstrip_outlets
    return get_devices(device_types.POWER_STRIP)
  File "/home/w1ll1am/.virtualenvs/hass_dev/config/deps/pywink/api.py", line 215, in get_devices
    response_dict = wink_api_fetch()
  File "/home/w1ll1am/.virtualenvs/hass_dev/config/deps/pywink/api.py", line 204, in wink_api_fetch
    response = requests.get(arequest_url, headers=API_HEADERS)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/api.py", line 70, in get
    return request('get', url, params=params, **kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/api.py", line 56, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/sessions.py", line 488, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/sessions.py", line 609, in send
    r = adapter.send(request, **kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/adapters.py", line 473, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', BadStatusLine("''",))
17-01-02 17:26:17 homeassistant.components.binary_sensor: Error while setting up platform wink
Traceback (most recent call last):
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 594, in urlopen
    chunked=chunked)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 391, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 387, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.4/http/client.py", line 1227, in getresponse
    response.begin()
  File "/usr/lib/python3.4/http/client.py", line 386, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.4/http/client.py", line 356, in _read_status
    raise BadStatusLine(line)
http.client.BadStatusLine: ''

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/adapters.py", line 423, in send
    timeout=timeout
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 643, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/util/retry.py", line 334, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/packages/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 594, in urlopen
    chunked=chunked)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 391, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/packages/urllib3/connectionpool.py", line 387, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.4/http/client.py", line 1227, in getresponse
    response.begin()
  File "/usr/lib/python3.4/http/client.py", line 386, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.4/http/client.py", line 356, in _read_status
    raise BadStatusLine(line)
requests.packages.urllib3.exceptions.ProtocolError: ('Connection aborted.', BadStatusLine("''",))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/homeassistant-0.36.0.dev0-py3.4.egg/homeassistant/helpers/entity_component.py", line 150, in _async_setup_platform
    entity_platform.add_entities, discovery_info
  File "/usr/lib/python3.4/asyncio/futures.py", line 358, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.4/asyncio/tasks.py", line 297, in _wakeup
    future.result()
  File "/usr/lib/python3.4/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.4/concurrent/futures/thread.py", line 54, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/homeassistant-0.36.0.dev0-py3.4.egg/homeassistant/components/binary_sensor/wink.py", line 43, in setup_platform
    for hub in pywink.get_hubs():
  File "/home/w1ll1am/.virtualenvs/hass_dev/config/deps/pywink/api.py", line 186, in get_hubs
    return get_devices(device_types.HUB)
  File "/home/w1ll1am/.virtualenvs/hass_dev/config/deps/pywink/api.py", line 215, in get_devices
    response_dict = wink_api_fetch()
  File "/home/w1ll1am/.virtualenvs/hass_dev/config/deps/pywink/api.py", line 204, in wink_api_fetch
    response = requests.get(arequest_url, headers=API_HEADERS)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/api.py", line 70, in get
    return request('get', url, params=params, **kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/api.py", line 56, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/sessions.py", line 488, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/sessions.py", line 609, in send
    r = adapter.send(request, **kwargs)
  File "/home/w1ll1am/.virtualenvs/hass_dev/lib/python3.4/site-packages/requests-2.12.4-py3.4.egg/requests/adapters.py", line 473, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', BadStatusLine("''",))
```

### PR DESCRIPTION
**Pull request by w1ll1am23**
There is some much going on here, I don't know if I can even remember what all is different. I'll do my best to explain.

 - First of all, and this is just my opinion, but splitting unique devices into their own file seems like a good idea. We were doing this with light_bulbs and thermostats so I went all out.
- Only API code is in the api.py
- The API is only called once when calling get_* the JSON is now stored locally (for 60 seconds), should speed up the Home-Assistant startup, and might fix #64 
- Using a default user-agent (Makes more details come back in the JSON!)
- objectprefix isn't used anymore because when using the above user-agent we get "object_type" and "object_id" which are generic fields that give us everything we need to know to query the API.
- Power strips now return three devices. Two outlets, and the entire power strip. The main power strip device turns both switches on/off at once. This fixes issues I have seen when turning both on at once and only one switch coming on.
- Support for lutron connected bulb remotes
- sensors_pods, smokedetectors, and door_bells, are now one object which reviews the capabilities to determine the sensor type. Much cleaner way to get the sensor, and will also make adding more sensors easier.
- Can now get all devices at once (helps with testing, there may be other benefits)

Most, if not all tests will fail currently because the JSON is bad and I haven't changed any of it for these changes. I think coverage should increase and tests should be easier to write now though?

This addresses #62 #65 and #69 

Validated that device is created and state can be queried. 
- [x]   light bulbs (RGB, temperature only, brightness only)
- [x]   Sensors (Door/Window, Motion/Temperature, Spotter)
- [x]  power strip
- [x]  eggminder
- [x]  lock and keys
- [x]  binary switch (zwave switch)
- [x]  Lutron remote
- [x]  thermostat
- [x]  Hub (v1)
- [x]  Siren
- [x]  porkfolio
- [X]  liquid sensor
- [X] garage doors
- [X] shades
- [X]  fan
- [X] door bell
- [X] smoke detector
- [X] Wink relay switches
- [X] Wink relay buttons
- [X] Wink relay sensors
- [X] Sprinkler

Test scripts have been written for the following
- [x]   light bulbs (RGB, temperature only, brightness only)
- [X]   Sensors (Door/Window, Motion/Temperature, Spotter)
- [x]  power strip
- [x]  eggminder
- [x]  lock
- [x] keys
- [x]  binary switch (zwave switch)
- [x]  Lutron remote
- [X]  thermostat
- [x]  Hub (v1, v2, hue hub)
- [x]  Siren
- [x]  porkfolio
- [x]  liquid sensor
- [x] garage doors
- [x] shades
- [x]  fan
- [x] door bell
- [x] smoke detector
- [X] Wink relay switches
- [x] Wink relay sensors
- [x] Sprinkler
- [x] Camera

Small script I am using to test.

```python
import pywink
import time

pywink.set_bearer_token('YOUR TOKEN')

wink_devices = pywink.get_all_devices()
for device in wink_devices:
    print(device.name())
    print(str(device.state()))
    device.update_state()
    time.sleep(2)
    print(device.name())
    print(str(device.state()))
```

### CODE REVIEW

**Review by bradsk88 — APPROVED** [review]
Big PR but it all looks fine to me.

Please update the version to 1.0.0.
1) There are some changes here that change the interface
2) I've been meaning to bump this out of 0-land for a while
3) The library has been integrated with homeassistant for some time now and has proven to be reasonably stable

**Comment by bradsk88** [comment]
You have merge permissions now.  So, feel free to merge once you've finished testing.

**Comment by w1ll1am23** [comment]
@scadaguru Any chance you could run the above test script and verify that your door bell sensors still show up? I know we just added support, but this PR really changes the way things are handled. Thanks.

Also, if possible could you provide me with your door bell JSON again, but this time get it will a user-agent of "Manufacturer/python-wink python/3 Wink/3" or from [wink@home](https://winkathome.net/Login.aspx?redirect=default.aspx)

Going to try to get updated versions of all the JSON we use for testing using the new user-agent.

**Comment by scadagenius** [comment]
Sure, I will test that script and provide the results tomorrow. Which code/repository I need to pull/get again to test?
I can test wink garage too if you want. 
About JSON, not sure what do I need to do different from last time when I provided JSON? You said user-agent not sure what that mean? Last time I did using similar to wink@home. 
I will do this tomorrow.

**Comment by w1ll1am23** [comment]
@scadaguru So are you saying you pulled the JSON from wink@home? Based on the JSON I don't think that is the case. git pull from here https://github.com/w1ll1am23/python-wink and git checkout switch_to_object_type. If you run the above script it will test all devices in your account.

**Comment by scadagenius** [comment]
Here is the JSON from wink@home for Ring Doorbell. I will test script later today and share the results.

```
{
  "object_type": "door_bell",
  "object_id": "xx",
  "uuid": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
  "icon_id": null,
  "icon_code": null,
  "desired_state": {},
  "last_reading": {
    "battery": 1.0,
    "battery_updated_at": 1483588707.0458965,
    "motion": false,
    "motion_updated_at": 1484327164.8382845,
    "button_pressed": false,
    "button_pressed_updated_at": 1484257789.6575472,
    "motion_true": "N/A",
    "motion_true_updated_at": 1484327124.6115115,
    "button_pressed_true": "N/A",
    "button_pressed_true_updated_at": 1484257783.5323548,
    "connection": true,
    "connection_updated_at": 1442873339.2134194,
    "last_recording_cuepoint_id": "1790157958",
    "last_recording_cuepoint_id_updated_at": 1484327268.7613738,
    "motion_changed_at": 1484327164.8382845,
    "motion_true_changed_at": 1484327124.6115115,
    "button_pressed_changed_at": 1484257789.6575472,
    "button_pressed_true_changed_at": 1484257783.5323548,
    "battery_changed_at": 1483588707.0458965,
    "last_recording_cuepoint_id_changed_at": 1484327268.7613738
  },
  "subscription": {
    "pubnub": {
      "subscribe_key": "sub-x-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
      "channel": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx|door_bell-xx|user-xxxxxx"
    }
  },
  "door_bell_id": "xx",
  "name": "xxxx",
  "locale": "en_us",
  "units": {},
  "created_at": 1442873339,
  "hidden_at": null,
  "capabilities": {
    "fields": [
      {
        "type": "percentage",
        "field": "battery",
        "mutability": "read-only"
      },
      {
        "type": "boolean",
        "field": "motion",
        "mutability": "read-only"
      },
      {
        "type": "boolean",
        "field": "button_pressed",
        "mutability": "read-only"
      }
    ]
  },
  "manufacturer_device_model": "doorbell",
  "manufacturer_device_id": "xxxxx",
  "device_manufacturer": "ring",
  "model_name": "Ring Video Doorbell",
  "upc_id": "xxx",
  "upc_code": "xxxxxxxxxxxx",
  "linked_service_id": "xxxxxx",
  "lat_lng": [
    xx.xxxxxx,
    xx.xxxxxx
  ],
  "location": "1234 Main St, City, ST XXXXX, USA"
}

```

**Comment by w1ll1am23** [comment]
@scadaguru perfect, thanks. Yeah the first JSON you provided was missing the object_type field.

**Comment by scadagenius** [comment]
Yeah, I used some Java based tool to authorized and pull the connected devices but didn't had that information weird though why JSON should differ if it is coming from the Wink Hub(#2) as poll/get results?

**Comment by w1ll1am23** [comment]
Yes, it is weird. This PR adds the user-agent that is required in order to return the correct fields. I think it is probably something strange going on in the Wink API but their support rarely helps out developers so I don't ask and/or tell them about issues very often.

**Comment by scadagenius** [comment]
By the way off the topic but I have Rachio sprinkler connected to Wink Hub and that shows status of watering of different zones as well schedules. Rachio has API available but still missing HA component, do you think it's good idea to bring Rachio to HA via Wink hub? Just asking.

**Comment by w1ll1am23** [comment]
@scadaguru if it pairs with Wink I would like to add support for it. My goal is to support every compatible device. Can you open a new issue on this project and include the JSON from wink@home for it?

**Comment by scadagenius** [comment]
Done created an issue #69 and attached JSON.

**Comment by w1ll1am23** [comment]
@scadaguru This PR has sprinkler support now too, so if you could test that when you are testing the door bell that would be sweet.

File: src/pywink/devices/sprinkler.py
Comment (bradsk88) [inline src/pywink/devices/sprinkler.py]: Docstring is wrong :stuck_out_tongue:

**Comment by scadagenius** [comment]
I have tested the test script above mentioned and doorbell showed up as expected but it didn't find the Rachio sprinkler. Let me know anything else if you want to try out.

**Comment by scadagenius** [comment]
I checked few things regarding sprinkler: Wink Android App allows to just make it to standby mode or online mode. JSON of Rachio has one controllable (read+write) and one read-only capability.
```
"capabilities": {
    "fields": [
      {
        "type": "boolean",
        "field": "connection",
        "mutability": "read-only"
      },
      {
        "type": "boolean",
        "field": "powered",
        "mutability": "read-write"
      }
    ]
  },

```

**Comment by w1ll1am23** [comment]
@scadaguru sorry, forgot more stuff again. Try again.

**Comment by scadagenius** [comment]
Perfect, now I see it. 
That was quick, thanks.

**Comment by scadagenius** [comment]
By the way please let me know when you update HA component to add these changes so I can spin-up test instance of HA on new machine to verify it works!

**Comment by w1ll1am23** [comment]
trust me i will. thanks for the help.

**Review by bradsk88 — APPROVED** [review]
Checking to see if this needs to be approved before it can be merged

**Comment by w1ll1am23** [comment]
I have done all of the testing I can. Going to merge this so it can be pushed to pypi and I can get more people to test it more easily.

**Pull request merged**
